gloss/Graphics/Gloss/Geometry/Triangulation.hs

```diff
@@ -0,0 +1,388 @@
+module Graphics.Gloss.Geometry.Triangulation
+        (triangulate)
+where
+        --TODO put exported functions here. triangulate only to be exact. pozdrav!
+import Graphics.Gloss.Rendering
+import Graphics.Gloss.Geometry.Line
+import Graphics.Gloss.Geometry.Angle
+import Graphics.Gloss.Data.Vector
+import qualified Data.Set as Set
+import qualified Data.Map.Strict as Map
+import Data.List (minimumBy, maximumBy, sortBy)
+import Control.Applicative
+import Data.Maybe
+
+-- import Debug.Trace
+
+
+-- | Type for representing planar graphs.
+type Vertices = Map.Map Vertex (Set.Set Vertex)
+
+-- | Whrepper for @Point@. Has different @Eq@ implmentation to absorb numerical error in calculations.
+data Vertex
+        = Vertex Point
+        deriving (Show)
+
+-- | Utility function to convert @Vertex@ to @Point@
+toPoint :: Vertex -> Point
+
+toPoint (Vertex x) = x
+
+instance Eq Vertex where
+        (Vertex (x1,y1)) == (Vertex (x2,y2)) = abs (x1-x2) + abs (y1-y2) < 0.0001
```

Comment:
Is there way to reformulate the algorithm so that it does not need this constant?